packages/solarwinds-apm/src/api.ts

```diff
@@ -24,8 +29,21 @@ import { SAMPLER } from "./shared/init.js"
  * @param timeout - Wait timeout in milliseconds
  * @returns Whether the library is ready
  */
-export function waitUntilReady(timeout: number): Promise<boolean> {
-  return SAMPLER.then((sampler) => sampler.waitUntilReady(timeout))
+export async function waitUntilReady(timeout: number): Promise<boolean> {
+  const sampler = await SAMPLER
+  return await sampler.waitUntilReady(timeout)
+}
+
+/**
+ * Forces the library to flush any buffered traces, metrics or logs
+ */
+export async function forceFlush(): Promise<void> {
+  const providers = await Promise.all([
+    TRACER_PROVIDER,
+    METER_PROVIDER,
+    LOGGER_PROVIDER,
+  ])
+  await Promise.all(providers.map((provider) => provider?.forceFlush()))
```

Comment:
I think `provider?` handles the case of library disabled/no-op mode, correct?

Reply:
It handles the case where `exportLogsEnabled` is `false` and no logger provider is registered

Reply:
I see. Would there be a library disabled/no-op scenario where this API call throws?

Reply:
Mhm, now that I think about it in that case it would probably hang forever.